src/cli/common.rs

```diff
@@ -397,7 +399,9 @@ pub(crate) fn list_targets(distributable: DistributableToolchain<'_>) -> Result<
 pub(crate) fn list_installed_targets(
     distributable: DistributableToolchain<'_>,
 ) -> Result<utils::ExitCode> {
-    let mut t = term2::stdout();
+    let t = process().stdout();
+    let mut t_lock = t.lock();
```

Comment:
We should probably not lock around non-console IO; instead locking just before the for loop.